### ISSUE TITLE
Allow start_tls to specify ciphers / ssl version(s)

### ISSUE THREAD

**Issue by emschwar** [eventmachine/eventmachine#603]
Right now, start_tls loads all OpenSSL algorithms, including the deprecated RC4. It would be nice if we could pass in a list of ciphers to `start_tls`, a la:

```
start_tls(
  private_key_file: CERT_KEY,
  cert_chain_file: CERT_CHAIN,
  verify_peer: false,
  ciphers: %w(ECDHE-RSA-AES256-GCM-SHA384 ECDHE-ECDSA-AES256-GCM-SHA384 etc etc)
)
```

It might be nice to also specify SSL versions-- for instance the ability to disable SSLV2 and enable only TLS 1.2 or higher, etc.

**Comment by sodabrew** [comment]
Yep, this is on the wishlist.

**Comment by dtshepherd** [comment]
+1 for this request, I was using https://github.com/ibc/EventMachine-LE for ssl version support due to failing SSL handshakes with an old web service.  But now I need the fixes from https://github.com/eventmachine/eventmachine/commit/8b613d05047ad6954f7d856b1404522214e8f001 (fix for #441 and #445).

**Comment by sodabrew** [comment]
"This year," I think. I've been working through bugs and internal cleanups in 1.0.x releases, so the codebase is in good shape.

Community help to move the eventmachine-le integration forward greatly appreciated, e.g. reviewing commits to see what needs to be ported.

**Comment by BarthV** [comment]
:+1: 
It would be awesome to allow EM talking to any kind of SSL/TLS ;o
Currently we're stuck because of this : https://github.com/seegno/sensu-influxdb-extension/pull/27

**Comment by sodabrew** [comment]
@dtshepherd The SSL Handshake failure fix will be in eventmachine 1.0.9. Can you test the eventmachine git `master` branch to let me know if it's working as expected now?

**Comment by BarthV** [comment]
Thanks 
I will test that !

**Comment by sodabrew** [comment]
Implemented in #654

**Comment by BarthV** [comment]
Thanks :+1:  :) !
